File: java/1.7/factory/src/com/pizza/pizzaStore/ChicagoPizzaStore.java
package com.pizza.pizzaStore;

import com.pizza.pizza.CheesePizza;
import com.pizza.pizza.ClamPizza;
import com.pizza.pizza.PepperoniPizza;
import com.pizza.pizza.Pizza;
import com.pizza.pizza.VeggiePizza;
import com.pizza.pizzaIngredientFactory.ChicagoPizzaIngredientFactory;
import com.pizza.pizzaIngredientFactory.PizzaIngredientFactory;

public class ChicagoPizzaStore extends PizzaStore {

	protected Pizza createPizza(String item) {
		Pizza pizza = null;
		PizzaIngredientFactory ingredientFactory =
		new ChicagoPizzaIngredientFactory();

		if (item.equals("cheese")) {

			pizza = new CheesePizza(ingredientFactory);
			pizza.setName("Chicago Style Cheese Pizza");

		} else if (item.equals("veggie")) {

			pizza = new VeggiePizza(ingredientFactory);
			pizza.setName("Chicago Style Veggie Pizza");

		} else if (item.equals("clam")) {

			pizza = new ClamPizza(ingredientFactory);
			pizza.setName("Chicago Style Clam Pizza");

		} else if (item.equals("pepperoni")) {

			pizza = new PepperoniPizza(ingredientFactory);
			pizza.setName("Chicago Style Pepperoni Pizza");

		}
		return pizza;
	}
}

File: python/2.7/factory/com/pizza/pizzaStore/ChicagoPizzaStore.py
import sys

from PizzaStore import *

sys.path.append("./pizzaIngredientFactory")
from ChicagoPizzaIngredientFactory import *

sys.path.append("./pizza")
from CheesePizza import *
from VeggiePizza import *
from ClamPizza import *
from PepperoniPizza import *

class ChicagoPizzaStore(PizzaStore):
    def createPizza(self, item):
        pizza = None
        ingredientFactory = ChicagoPizzaIngredientFactory()

        if item == "cheese":
            pizza = CheesePizza(ingredientFactory)
            pizza.setName("Chicago Style Cheese Pizza")
        elif item == "veggie":
            pizza = VeggiePizza(ingredientFactory)
            pizza.setName("Chicago Style Veggie Pizza")
        elif item == "clam":
            pizza = ClamPizza(ingredientFactory)
            pizza.setName("Chicago Style Clam Pizza")
        elif item == "pepperoni":
            pizza = PepperoniPizza(ingredientFactory)
            pizza.setName("Chicago Style Pepperoni Pizza")
        return pizza